tests/test_formatter.py

```diff
@@ -178,3 +183,62 @@ def test_internal_exception_formatting(caplog: pytest.LogCaptureFixture):
     assert len(caplog.records) == 1
     assert caplog.records[0].message.startswith('Caught an internal error in Logfire.')
     assert str(caplog.records[0].exc_info[1]) == 'bad scrubber'  # type: ignore
+
+
+@pytest.mark.skipif(sys.version_info[:2] == (3, 8), reason='fstring magic is only for 3.9+')
+def test_syntax_error_in_ast_walk(monkeypatch: pytest.MonkeyPatch):
+    """Test the SyntaxError exception handling in ast.walk when checking for await expressions."""
+
+    def mock_ast_walk(*args: object, **kwargs: object) -> None:
+        raise SyntaxError('Simulated syntax error in ast.walk')
+
+    monkeypatch.setattr(ast, 'walk', mock_ast_walk)
+
+    node = ast.FormattedValue(value=ast.Name(id='test', ctx=ast.Load()), conversion=-1, format_spec=None)
+
+    class MockSource:
+        text = 'test'
+
+    # This should not raise an exception because the SyntaxError is caught
+    from logfire._internal.formatter import compile_formatted_value
+
+    _, value_code, formatted_code = compile_formatted_value(node, MockSource())
+
+    assert isinstance(value_code, CodeType)
+    assert isinstance(formatted_code, CodeType)
+
+
+@pytest.mark.skipif(sys.version_info[:2] == (3, 8), reason='fstring magic is only for 3.9+')
+def test_await_in_fstring(exporter: TestExporter):
```

Comment:
```suggestion
@pytest.mark.anyio
async def test_await_in_fstring(exporter: TestExporter):
```

Then you can just use await and not start an event loop. See the tests in test_logfire for example.